provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -735,6 +735,21 @@ func generateSchema() schema.PackageSpec {
 							"(https://docs.aws.amazon.com/eks/latest/APIReference/API_Nodegroup.html#AmazonEKS-Type-Nodegroup-amiType) " +
 							"for valid AMI Types. This provider will only perform drift detection if a configuration value is provided.",
 					},
+					"amiId": {
```

Comment:
You'll need to run `make build` and check in the changes to the schema and `sdk` folder. Otherwise those don't take effect.

Because of the disparity between the nodejs SDK and the rest of the SDKs it would be good to add a test in another language as well (e.g. Golang). Let me know if you need help with converting the test you've created into another language